src/unsafe-keyword.md

```diff
@@ -29,7 +29,7 @@ By putting operations into an unsafe block, the programmer states that they have
 Unsafe blocks are the logical dual to unsafe functions:
 where unsafe functions define a proof obligation that callers must uphold, unsafe blocks state that all relevant proof obligations have been discharged.
 There are many ways to discharge proof obligations;
-for example, there could be run-time checks or data structure invariants that guarantee that certain properties are definitely true, or the unsafe block could be inside an `unsafe fn` and use its own proof obligations to discharge the proof obligations of its callees.
+for example, there could be run-time checks or data structure invariants that guarantee that certain properties are definitely true, or the unsafe block could be inside an `unsafe fn` and use its own proof obligations to discharge the proof obligations of its callers.
```

Comment:
Hm, yea, the use of "callee" doesn't seem right, at least with my understanding in a caller/callee relationship of a function, the callee is the function itself. 

However, I've had to re-read this sentence about a dozen times to try to make sure it is clear in my head and I'm thinking some of my struggle is the use of "its" twice. That makes me need to think a little harder to understand which "it" is being referred to, and the relationship.

I'm wondering, does the following seem clearer to you?

```suggestion
for example, there could be run-time checks or data structure invariants that guarantee that certain properties are definitely true, or the unsafe block could be inside an `unsafe fn` and the unsafe block can use the safety requirements the unsafe function imposes on its callers to discharge the proof obligations of the unsafe block.
```

Reply:
I'm not totally sure that's better, because the last part references `proof obligations of the unsafe block`, when really I believe the sentence is trying to address the `proof obligations of the callers`, and saying that the unsafe block discharges **their** proof obligations. In other words, normally you have (with an `unsafe fn`):

1. Safety requirements of `unsafe fn`, which normally are met by the:
2. Proof obligation of the caller(s)

In the case of using an unsafe block inside an `unsafe fn` we have an additional:

3. Proof obligation of the unsafe block, which discharges the proof obligation of the callers (2).

Also, the unsafe block would not be using `the safety requirements the unsafe function imposes on its callers`; rather, it's using its own proof obligations.

We could replace a couple of the "its" like this:

```suggestion
for example, there could be run-time checks or data structure invariants that guarantee that certain properties are definitely true, or the unsafe block could be inside an `unsafe fn` and use the block's own proof obligations to discharge the proof obligations of the `unsafe fn`'s callers.
```